### PR TITLE
Scratch-style toolbox

### PR DESCRIPTION
**Pull request by rachel-fenichel**
- Vertical toolbox
- Horizontal toolbox is the same as before for simple, and with categories looks funny right now
- Small problem with differing scale between block menu and workspace

### CODE REVIEW

File: core/toolbox.js [hunk 1]
```diff
@@ -460,219 +276,177 @@ Blockly.Toolbox.prototype.getClientRect = function() {
...
+      case 'CATEGORY':
+        if (!(this.categories_.length % 2)) {
+          var row = goog.dom.createDom('tr', 'scratchCategoryMenuRow');
+          this.table.appendChild(row);
+        }
+        this.categories_.push(new Blockly.Toolbox.Category(this, row,
+            child));
+        break;
+      case 'SEP':
+        // TODO: deal with separators.
```
Comment (tmickel) [inline core/toolbox.js:355]: not sure if we actually need separators in the toolbox, btw!

Comment (rachel-fenichel) [reply core/toolbox.js:355]: Yup, I talked to @carljbowman and he said we don't need them.

**Comment by tmickel** [comment]
Discussed offline! Although this code is still a heroic work-in-progress, we decided that we'll merge it now to start exposing bugs early in the process.

**Comment by thisandagain** [comment]
🎉 🐱 🍰 💥

**Pull request merged**